lib/ddtrace/profiling/events/stack.rb

```diff
@@ -33,10 +33,8 @@ def initialize(
             thread_id,
             trace_id,
             span_id,
-            [
-              frames.collect(&:hash),
-              total_frame_count
-            ]
+            frames.collect(&:hash),
+            total_frame_count
```

Comment:
These are not equivalent (and there were no test changes), is this ok?

Reply:
I believe they are equivalent semantically -- the objective of lines 32 to 38 is to generate a single hash value that is a "fingerprint" for this stack. This is later used for a kind of aggregation de-duping -- if we have multiple events with the same stack, we sum their metric (wall/cpu-time) and represent it in the profiling data as a single stack with the sum of all the metrics from the events with the same fingerprint.

Thus, even though removing the array here means that the resulting hash value _will_ be different, I think it's semantically equivalent because the exact same data is being hashed. E.g., for our purposes, hashing [1, [2, [3]]] is equivalent to hashing [1, 2, 3] as long as we do the exact same transformation for every stack.

(I think I'm not missing something, but let me know!)

Reply:
I understand now, thank you!

Reply:
Thanks for making me explain! Always helps in making sure I'm thinking correctly about it :)